zap/src/main/java/org/zaproxy/zap/extension/history/SiteNotesAddDialog.java

```diff
@@ -0,0 +1,348 @@
+        super(arg0, arg1);
+        initialize();
+    }
+
+    /** This method initializes this */
+    private void initialize() {
+        this.setTitle(Constant.messages.getString("history.addnote.title"));
+
+        JPanel panel = new JPanel();
+        GroupLayout layout = new GroupLayout(panel);
+        panel.setLayout(layout);
+        layout.setAutoCreateGaps(true);
+        layout.setAutoCreateContainerGaps(true);
+
+        Component buttonsGlue = Box.createHorizontalGlue();
+
+        layout.setHorizontalGroup(
+                layout.createParallelGroup()
+                        .addGroup(
+                                layout.createSequentialGroup()
+                                        .addComponent(
+                                                getHrefSelectorListPanel(),
+                                                GroupLayout.DEFAULT_SIZE,
+                                                GroupLayout.PREFERRED_SIZE,
+                                                GroupLayout.PREFERRED_SIZE)
+                                        .addComponent(
+                                                getJScrollPane(),
+                                                GroupLayout.DEFAULT_SIZE,
+                                                GroupLayout.PREFERRED_SIZE,
+                                                Short.MAX_VALUE))
+                        .addGroup(
+                                layout.createSequentialGroup()
+                                        .addComponent(
+                                                buttonsGlue,
+                                                GroupLayout.DEFAULT_SIZE,
+                                                GroupLayout.PREFERRED_SIZE,
+                                                GroupLayout.DEFAULT_SIZE)
+                                        .addComponent(getBtnCancel())
+                                        .addComponent(getBtnOk())));
+
+        layout.setVerticalGroup(
+                layout.createSequentialGroup()
+                        .addGroup(
+                                layout.createParallelGroup()
+                                        .addComponent(getHrefSelectorListPanel())
+                                        .addComponent(getJScrollPane()))
+                        .addGroup(
+                                layout.createParallelGroup()
+                                        .addComponent(buttonsGlue)
+                                        .addComponent(getBtnCancel())
+                                        .addComponent(getBtnOk())));
+
+        setContentPane(panel);
+
+        this.addWindowListener(
+                new java.awt.event.WindowAdapter() {
+
+                    @Override
+                    public void windowClosing(java.awt.event.WindowEvent e) {
+                        clearAndDispose();
+                    }
+                });
```

Comment:
This could use a lambda I believe.

Reply:
Kinda like down at 188

Reply:
> This could use a lambda I believe.

lambda one is giving error i tried 
 ```
       this.addWindowListener((e)->{
            clearAndDispose();
        });
```

Reply:
Hmm. Anyone else got thoughts?

Leave it as-is for now, we can always come back to it.

Reply:
Needs to be that way (though the packages could be removed).